### PR TITLE
Replace default_*_transforms for a default_transforms preprocess hook

### PR DESCRIPTION
**Pull request by ethanwharris**
## What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/custom_task.rst [hunk 1]
```diff
@@ -232,20 +232,7 @@ Example::
...
-            return self.to_tensor
-
-        @property
-        def default_test_transforms(self) -> Optional[Dict[str, Callable]]:
-            return self.to_tensor
-
-        @property
-        def default_predict_transforms(self) -> Optional[Dict[str, Callable]]:
+        def default_transforms(self) -> Optional[Dict[str, Callable]]:
             return self.to_tensor
```
Comment (tchaton) [inline docs/source/custom_task.rst:236]:
```suggestion
        def default_transforms(self) -> Optional[Dict[str, Callable]]:
            # this default transform will be used for train, test, val and predict.
            return self.to_tensor
```

File: flash/data/process.py [hunk 2]
```diff
@@ -217,30 +223,16 @@ def __init__(
...
-
-    @property
-    def transforms(self) -> Dict[str, Optional[Dict[str, Callable]]]:
-        return {
-            "train_transform": self.train_transform,
-            "val_transform": self.val_transform,
-            "test_transform": self.test_transform,
-            "predict_transform": self.predict_transform,
-        }
+        with CurrentRunningStageFuncContext(running_stage, "default_transforms", self):
```
Comment (tchaton) [inline flash/data/process.py:233]: I think only CurrentRunningStageContext should be fine. Do we need current_fn = `default_transforms` ?

File: flash/data/transforms.py [hunk 3]
```diff
@@ -89,3 +89,31 @@ def kornia_collate(samples: Sequence[Dict[str, Any]]) -> Dict[str, Any]:
             if torch.is_tensor(sample[key]):
                 sample[key] = sample[key].squeeze(0)
     return default_collate(samples)
+
+
+def merge_transforms(
+    base_transforms: Dict[str, Callable],
+    additional_transforms: Dict[str, Callable],
```
Comment (tchaton) [inline flash/data/transforms.py:96]: Should we add an override option ? Mind adding tests ?

**Review by tchaton — COMMENTED** [review]
Really cool !

File: flash/vision/classification/transforms.py [hunk 4]
```diff
@@ -38,7 +40,6 @@ def default_train_transforms(image_size: Tuple[int, int]) -> Dict[str, Callable]
             "post_tensor_transform": ApplyToKeys(
                 DefaultDataKeys.INPUT,
                 K.geometry.Resize(image_size),
-                K.augmentation.RandomHorizontalFlip(),
```
Comment (edgarriba) [inline flash/vision/classification/transforms.py:41]: we need to run a retrain if we change this and update default weights in the examples

**Review by edgarriba — APPROVED** [review]

File: flash/vision/classification/transforms.py (hunk 4 above)
Comment (ethanwharris) [reply flash/vision/classification/transforms.py:41]: Haven't changed it, it just gets added in line 70

**Pull request merged**